pkg/controller/controlplane/valuesprovider.go

```diff
@@ -442,7 +451,10 @@ func getConfigChartValues(
 		// detect internal network.
 		// See https://github.com/kubernetes/cloud-provider-openstack/blob/v1.22.1/docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md#networking
 		"internalNetworkName": infraStatus.Networks.Name,
-		"routerID":            infraStatus.Networks.Router.ID,
+	}
+
+	if !isUsingOverlay {
+		values["routerID"] = infraStatus.Networks.Router.ID
```

Comment:
Will the deletion still work if the routerID is not there anymore? Because we made use of it in the delete operation in the actuator.

Reply:
The deletion uses the router.id returned from Terraformer. The terraformer still has the router id input (or the router is created). From my understanding of the code it is unrelated to the dropping of the routerID from the cloud provider config.